<a href="https://colab.research.google.com/github/rishi-29-02/Performing-Analysis-of-Meteorological-Data/blob/main/Weather_data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##Performing Analysis of Meteorological Data

Hypothesis to be tested : The Influence of Global Warming on temperature and humidity.

In [68]:
import pandas as pd

In [69]:
# load the dataset
#from google.colab import files
#uploaded = files.upload()

In [70]:
data = pd.read_csv('weatherHistory.csv')
data.head(8)

Formatted Date  ...                      Daily Summary
0  2006-04-01 00:00:00.000 +0200  ...  Partly cloudy throughout the day.
1  2006-04-01 01:00:00.000 +0200  ...  Partly cloudy throughout the day.
2  2006-04-01 02:00:00.000 +0200  ...  Partly cloudy throughout the day.
3  2006-04-01 03:00:00.000 +0200  ...  Partly cloudy throughout the day.
4  2006-04-01 04:00:00.000 +0200  ...  Partly cloudy throughout the day.
5  2006-04-01 05:00:00.000 +0200  ...  Partly cloudy throughout the day.
6  2006-04-01 06:00:00.000 +0200  ...  Partly cloudy throughout the day.
7  2006-04-01 07:00:00.000 +0200  ...  Partly cloudy throughout the day.

[8 rows x 11 columns]

In [71]:
data.shape

(96453, 11)

In [72]:
data.dtypes

Formatted Date               object
Summary                      object
Precip Type                  object
Temperature (C)             float64
Apparent Temperature (C)    float64
Humidity                    float64
Wind Speed (km/h)           float64
Wind Bearing (degrees)        int64
Visibility (km)             float64
Pressure (millibars)        float64
Daily Summary                object
dtype: object

In [73]:
data.describe(include='all')

Formatted Date  ...                      Daily Summary
count                           96453  ...                              96453
unique                          96429  ...                                214
top     2010-08-02 02:00:00.000 +0200  ...  Mostly cloudy throughout the day.
freq                                2  ...                              20085
mean                              NaN  ...                                NaN
std                               NaN  ...                                NaN
min                               NaN  ...                                NaN
25%                               NaN  ...                                NaN
50%                               NaN  ...                                NaN
75%                               NaN  ...                                NaN
max                               NaN  ...                                NaN

[11 rows x 11 columns]

From the describe we infer that the feature PrecipType has missing values.

In [74]:
data['Precip Type'].value_counts()

rain    85224
snow    10712
Name: Precip Type, dtype: int64

In [75]:
data['Precip Type'].isnull().value_counts()

False    95936
True       517
Name: Precip Type, dtype: int64

Thus, the feature Precip Type has 517 missing values.

Dropping all the instances having missing values.

In [76]:
data = data.dropna()

In [77]:
data.shape

(95936, 11)

In [78]:
data.describe(include='all')

Formatted Date  ...                      Daily Summary
count                           95936  ...                              95936
unique                          95912  ...                                214
top     2010-08-02 06:00:00.000 +0200  ...  Mostly cloudy throughout the day.
freq                                2  ...                              20020
mean                              NaN  ...                                NaN
std                               NaN  ...                                NaN
min                               NaN  ...                                NaN
25%                               NaN  ...                                NaN
50%                               NaN  ...                                NaN
75%                               NaN  ...                                NaN
max                               NaN  ...                                NaN

[11 rows x 11 columns]